File: VB/E1554.Module/NewViewFilterObjectController.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule

Namespace E1554.Module
	Public Class NewViewFilterObjectController
		Inherits ObjectViewController(Of ListView, ViewFilterObject)

		Public Sub New()
			TargetViewId = "ViewFilterObject_LookupListView"
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			If TypeOf View.CollectionSource Is PropertyCollectionSource AndAlso CType(View.CollectionSource, PropertyCollectionSource).MasterObjectType Is GetType(ViewFilterContainer) Then
				Dim newObjectViewController As NewObjectViewController = Frame.GetController(Of NewObjectViewController)()
				If newObjectViewController IsNot Nothing Then
					AddHandler newObjectViewController.ObjectCreated, AddressOf ViewController1_ObjectCreated
				End If
			End If
		End Sub
		Private Sub ViewController1_ObjectCreated(ByVal sender As Object, ByVal e As ObjectCreatedEventArgs)
			If TypeOf e.CreatedObject Is ViewFilterObject Then
				Dim newViewFilterObject As ViewFilterObject = CType(e.CreatedObject, ViewFilterObject)
				Dim pcs As PropertyCollectionSource = CType(View.CollectionSource, PropertyCollectionSource)
				newViewFilterObject.ObjectType = CType(pcs.MasterObject, ViewFilterContainer).ObjectType
			End If
		End Sub
		Protected Overrides Sub OnDeactivated()
			MyBase.OnDeactivated()
			If TypeOf View.CollectionSource Is PropertyCollectionSource AndAlso CType(View.CollectionSource, PropertyCollectionSource).MasterObjectType Is GetType(ViewFilterContainer) Then
				Dim newObjectViewController As NewObjectViewController = Frame.GetController(Of NewObjectViewController)()
				If newObjectViewController IsNot Nothing Then
					RemoveHandler newObjectViewController.ObjectCreated, AddressOf ViewController1_ObjectCreated
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/XPO/DialogBeforeListView/DialogBeforeListView.Module/Controllers/NewViewFilterObjectController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;

namespace E1554.Module {
    public class NewViewFilterObjectController : ObjectViewController<ListView, ViewFilterObject> {
        public NewViewFilterObjectController() {
            TargetViewId = "ViewFilterObject_LookupListView";
        }
        protected override void OnActivated() {
            base.OnActivated();
            if (View.CollectionSource is PropertyCollectionSource && ((PropertyCollectionSource)View.CollectionSource).MasterObjectType == typeof(ViewFilterContainer)) {
                NewObjectViewController newObjectViewController = Frame.GetController<NewObjectViewController>();
                if (newObjectViewController != null) { 
                    newObjectViewController.ObjectCreated += new EventHandler<ObjectCreatedEventArgs>(ViewController1_ObjectCreated);
                }     
            }
        }
        void ViewController1_ObjectCreated(object sender, ObjectCreatedEventArgs e) {
            if (e.CreatedObject is ViewFilterObject) {
                ViewFilterObject newViewFilterObject = (ViewFilterObject)e.CreatedObject;
                PropertyCollectionSource pcs = (PropertyCollectionSource)View.CollectionSource;
                newViewFilterObject.ObjectType = ((ViewFilterContainer)pcs.MasterObject).ObjectType;
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            if (View.CollectionSource is PropertyCollectionSource && ((PropertyCollectionSource)View.CollectionSource).MasterObjectType == typeof(ViewFilterContainer)) {
                NewObjectViewController newObjectViewController = Frame.GetController<NewObjectViewController>();
                if (newObjectViewController != null) { 
                    newObjectViewController.ObjectCreated -= new EventHandler<ObjectCreatedEventArgs>(ViewController1_ObjectCreated);
                }
            }
        }
    }
}
